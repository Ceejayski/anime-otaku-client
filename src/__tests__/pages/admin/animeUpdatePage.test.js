import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import AnimeUpdatePage from '../../../pages/admin/animeUpdatePage';
import store from '../../../redux';

test('renders Correctly', () => {
  const Placeholder = () => (
    <Provider store={store}>
      <BrowserRouter>
        <AnimeUpdatePage />
      </BrowserRouter>
    </Provider>
  );
  const tree = create(<Placeholder />).toJSON();
  expect(tree).toMatchSnapshot();
});
