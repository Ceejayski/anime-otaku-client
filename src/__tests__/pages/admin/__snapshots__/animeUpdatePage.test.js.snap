// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Correctly 1`] = `
<div>
  <form
    className="main-form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="card"
    >
      <h4
        className="text-center mb-3"
      >
        Create New Anime
      </h4>
      <div
        className="mb-3"
      >
        <label
          htmlFor="name"
        >
          Title
        </label>
        <input
          className="form-control "
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
        <small
          className="text-muted form-text"
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="form-control "
          cols="50"
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          rows="20"
        />
        <small
          className="text-muted form-text"
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="rating"
        >
          Rating
        </label>
        <input
          className="form-control "
          id="rating"
          name="rating"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <small
          className="text-muted form-text"
        />
      </div>
      <div
        className="mb-3 "
      >
        <label
          className="form-label"
        >
          Anime image
        </label>
        <input
          className="form-control"
          id="header_image"
          name="header_image"
          onBlur={[Function]}
          onChange={[Function]}
          type="file"
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
        >
          Genre list
          <div
            className="w-100 css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-1s2u09g-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-319lph-ValueContainer"
              >
                <div
                  className=" css-14el2xx-placeholder"
                  id="react-select-2-placeholder"
                >
                  Select...
                </div>
                <div
                  className=" css-6j8wv5-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-2-listbox"
                    aria-describedby="react-select-2-placeholder"
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-owns="react-select-2-listbox"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    className=""
                    disabled={false}
                    id="react-select-2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "color": "inherit",
                        "font": "inherit",
                        "gridArea": "1 / 2",
                        "label": "input",
                        "margin": 0,
                        "minWidth": "2px",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={0}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="genre_list"
                type="hidden"
              />
            </div>
          </div>
        </label>
      </div>
      <div
        className="form-group mx-auto"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Create
        </button>
        <a
          className="btn btn-link"
          href="/"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
    </div>
  </form>
</div>
`;
